project/driver_controls/src/main.c

```diff
@@ -0,0 +1,131 @@
+#include <stdio.h>
+#include <stdint.h>
+
+#include "adc.h"
+#include "input_interrupt.h"
+#include "driver_state.h"
+#include "soft_timer.h"
+#include "driver_device.h"
+
+#include "power_state.h"
+#include "pedal_state.h"
+#include "direction_state.h"
+#include "turn_signal_state.h"
+#include "hazard_light_state.h"
+#include "mechanical_brake_state.h"
+
+#define INPUT_DEVICES 16
+#define OUTPUT_DEVICES 1
+
+// Struct of FSMs to be used in the program
+typedef struct FSMGroup {
+  FSM power;
+  FSM pedal;
+  FSM direction;
+  FSM turn_signal;
+  FSM hazard_light;
+  FSM mechanical_brake;
+} FSMGroup;
+
+void device_init() {
+  driver_device_init();
+
+  // Configure driver devices with their individual settings
+  DriverDevice inputs[INPUT_DEVICES] = {
```

Comment:
I don't know if the `[ENUM] = { ... }` strategy would work with struct elements.